includes/class-newspack-popups-segmentation.php

```diff
@@ -417,6 +453,34 @@ function ( $client_data ) use ( $segment_config ) {
 		];
 	}
 
+	/**
+	 * Sort all segments by relative priority.
+	 *
+	 * @param object $segments Array of segments, sorted by priority property.
+	 */
+	public static function sort_segments( $segments ) {
+		update_option( self::SEGMENTS_OPTION_NAME, self::reindex_segments( $segments ) );
+		return self::get_segments();
+	}
+
+	/**
+	 * Reindex segment priorities based on current position in array.
+	 *
+	 * @param object $segments Array of segments.
+	 */
+	public static function reindex_segments( $segments ) {
+		$index = 0;
+		array_map(
+			function( $segment ) use ( $index ) {
+				$segment['priority'] = $index;
+				$index++;
+			},
+			$segments
+		);
+
+		return $segments;
```

Comment:
This didn't work for me because the `array_map` is not actually updating the `$segments` variable. Here's a solution:

```php
return array_map(
	function( $segment ) use ( &$index ) {
		$segment['priority'] = $index;
		$index++;
		return $segment;
	},
	$segments
);
```

(note the `&` before `$index` - it's there to make make the variable mutable)

Reply:
Ah, I always wondered what the `&$` meant. Fixed in 6a5f9ed, thanks.

Reply:
Still, the array passed as the argument to this method will not be updated - this can be verified by logging the value of `$segments` before and after the `array_map`.

Reply:
Oh, that's weird. How about if I use `array_values` instead—that will reindex an ordered array with the same values, right? Trying that out in `2958a5e`. Kind of hard to test without spoofing an array value in `get_segments`.

Reply:
This approach will not handle legacy segments, which do not have a `priority` key. 
This can be tested by `unset`ting a segment's `priority` right after option value retrieval in `get_segments` method. 
I'd still recommend the solution from the top comment.

Reply:
Ack, you're right. And I had missed a line from your first suggestion, so I thought it wasn't working. Reimplemented your solution without any missing code in 2b10b26 and it now seems to work for me 😄